packages/cozy-client/src/CozyClient.js

```diff
@@ -1157,7 +1157,18 @@ client.query(Q('io.cozy.bills'))`)
     if (!Array.isArray(data)) {
       await this.persistVirtualDocument(data, enforce)
     } else {
-      for (const document of data) {
+      const documentsToPersist = data.filter(document => {
+        if (!document || document.cozyLocalOnly) {
+          return false
+        }
+
+        if ((!document.meta?.rev && !document._rev) || enforce) {
+          return true
+        }
+
+        return false
+      })
+      for (const document of documentsToPersist) {
```

Comment:
Is it really better? OK you don't call a method if not needed, but now you loop twice

Reply:
This is actually significantly better. After a big query retrieving 25K docs, the `persistVirtualDocuments` takes: 
- Before: 609ms
- After: 20ms

I wonder if this is because of the event loop overwhelmed by too many calls

Reply:
Oh shit, I didn't saw that the `persistVirtualDocument` was async... This is why.

nit: You can now remove the check within the `persistVirtualDocument` method since should only call it with the docs to persist ^^.